lib/Parser/Parse.cpp

```diff
@@ -1398,15 +1398,52 @@ ParseNodePtr Parser::AddVarDeclNode(IdentPtr pid, ParseNodePtr pnodeFnc)
     return pnode;
 }
 
-ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr pid)
+Js::PropertyId Parser::EnsurePropertyId(IdentPtr pid)
 {
-    ParseNodePtr declNode = CreateBlockScopedDeclNode(pid, knopConstDecl);
+    Js::PropertyId propertyId = pid->GetPropertyId();
+    if (propertyId == Js::Constants::NoProperty)
+    {
+        propertyId = m_scriptContext->GetOrAddPropertyIdTracked(pid->Psz(), pid->Cch());
+        pid->SetPropertyId(propertyId);
+    }
+    return propertyId;
+}
 
-    declNode->sxVar.sym->SetIsModuleExportStorage(true);
+ParseNodePtr Parser::CreateModuleImportDeclNode(IdentPtr localName)
+{
+    ParseNodePtr declNode = CreateBlockScopedDeclNode(localName, knopConstDecl);
+    Symbol* sym = declNode->sxVar.sym;
+    
+    sym->SetIsModuleExportStorage(true);
 
     return declNode;
 }
 
+Symbol* Parser::MarkIdentifierReferenceIsModuleExport(IdentPtr localName)
+{
+    PidRefStack* pidRef = this->PushPidRef(localName);
```

Comment:
It's true that export statements can only appear at global scope, yes. But they can export any local name which is why I'm looking up the symbol for the name. I found in testing that the Symbol* in the PidRefStack* I got back from PushPidRef was nullptr but I was able to walk back along the stack and find a Symbol* which I could use to mark the name as export storage. 

We can have some syntax like this:
var foo = 1; 
function bar() { foo++ };
export { foo, bar };